### ISSUE TITLE
failed sandbox add: failed to add interface veth92b5e1e to sandbox: failed in prefunc: failed to set namespace on link "veth92b5e1e": invalid argument

### ISSUE THREAD

**Issue by sw1sh** [docker/docker#15772]
```
-> % docker version
Client:
 Version:      1.9.0-dev
 API version:  1.21
 Go version:   go1.4.2
 Git commit:   5fd15da
 Built:        Sat Aug 22 04:14:56 UTC 2015
 OS/Arch:      linux/amd64

Server:
 Version:      1.9.0-dev
 API version:  1.21
 Go version:   go1.4.2
 Git commit:   5fd15da
 Built:        Sat Aug 22 04:14:56 UTC 2015
 OS/Arch:      linux/amd64
```

```
-> % docker info
Containers: 22
Images: 28
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 72
 Dirperm1 Supported: true
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 3.19.0-27-generic
Operating System: Ubuntu 14.04.3 LTS
CPUs: 8
Total Memory: 15.31 GiB
Name: lenovo
ID: TA2Y:DHEU:5GOY:YHDX:E24K:C66E:H35N:JMJY:A2DM:NKGM:2OR4:MUOY
WARNING: No swap limit support
```

```
-> % uname -a
Linux lenovo 3.19.0-27-generic #29~14.04.1-Ubuntu SMP Sun Aug 16 01:51:13 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
```

Almost fresh Ubuntu installation, had a RHEL6.6 before with docker working fine there.
Tried script from https://get.docker.com/ and latest build, same error.

Basically can't start anything:
```
-> % docker run hello-world
Error response from daemon: Cannot start container ea8ded236cc909153d1afa9435094f4a34221d76408dd11a710629232fdd6214: failed sandbox add: failed to add interface veth324ba41 to sandbox: failed in prefunc: failed to set namespace on link "veth324ba41": invalid argument
```

**Comment by outcoldman** [comment]
I have the same issue with Ubuntu. If you are trying to run it multiple times - it can start on N run.
Another weird issue - that `docker stats --no-stream=true` can return all 0. It looks like that docker maybe timeouts or something...

```
Containers: 34
Images: 2025
Storage Driver: aufs
 Root Dir: /mnt/DOCKER/runtime/aufs
 Backing Filesystem: extfs
 Dirs: 2093
 Dirperm1 Supported: false
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 3.13.0-62-generic
Operating System: Ubuntu 14.04.3 LTS
CPUs: 8
Total Memory: 31.13 GiB
Name: outcoldbuntu
ID: YC6H:RMJL:LOH6:AVMY:6GPO:3AUZ:OU2J:IFTU:FNO5:I62N:GKIZ:ZKGT
Labels:
 provider=generic
```

**Comment by akoptelov** [comment]
The same issue with kernel 3.13.0-62 and docker 1.8.1.

Couldn't be some h/w issue? I see the sw1sh uses lenovo, and mine is W530.

PS. outcoldman, N is 500 and still no luck.

**Comment by BenHall** [comment]
Related to #15341

**Comment by cpuguy83** [comment]
Closing as dup of #15341

**Comment by mavenugo** [comment]
based on our investigation & available data, this Issue doesnt seem to be related to #15341. More info : https://github.com/docker/docker/issues/15341#issuecomment-136829495.
Reopening to get more info from @sw1sh

**Comment by mavenugo** [comment]
@sw1sh do you still see this issue ? Can you please respond to @mrjana's question : https://github.com/docker/docker/issues/15341#issuecomment-136776379 ?
We can close this issue if you dont see it anymore or cannot reproduce it.

**Comment by sw1sh** [comment]
I can't boot into *-26 kernel right now, video drivers won't work there for
some reason and it loads in low-graphics mode and docker daemon refuses to
start, complaining about it. On *-27 I still can't start a daemon.

**Comment by akoptelov** [comment]
@mavenugo @sw1sh Maybe this might help you. 
I tried to reproduce the problem on KVM machines. One is plain 14.04, with 3.19 kernel, the other is 14.04 with some additional packages required in our company and kernel 3.13. Docker runs just fine on the former one and gives 'invalid argument' on later one. Installing 3.19 kernel makes Docker happy though.

Stopping services specific to our company doesn't change the situation. Also upgrading my host to 3.19 kernel doesn't fix Docker.

Does it give any clue for you? Is any additional info needed?

**Comment by mrjana** [comment]
@akoptelov You may be having a different problem. Can you please post your exact error message?

Also in the host where you getting this error can you see if you are running Symantec AV? The service name is `savap`. If so can you bring that down and try?

**Comment by akoptelov** [comment]
@mrjana May be. Still the synopsis is the same:

    Error response from daemon: Cannot start container 82650cdd5fec3e79d4236c48e82c0b7ca34eaca4de8ad891aef5780dc2e67dc1: failed sandbox add: failed to add interface veth7f3634c to sandbox: failed in prefunc: failed to set namespace on link "veth7f3634c": invalid argument

As to the Symantec AV, I believe the service is autoprotect. And I get this error with it either up or down.

**Comment by lucasvel** [comment]
I think we are all having this issue at IBM with our ThinkPads.
I've found a solution to this. I had to disable the SAV autoprotect from booting:
```
sudo update-rc.d autoprotect disable
```
After that, reboot your ThinkPads and try docker commands.

**Comment by akoptelov** [comment]
@lucasvel Thanks, that makes docker work. I guess it wasn't enough to stop autoprotect. Instead it should be turned off ab initio. Still this doesn't completely solves the problem for ours IBMers. autoprotect is mandatory for running. @mrjana FYI.

**Comment by mrjana** [comment]
@akoptelov @lucasvel This problem is known for some time now. May be we should add a check in the config checking script. But unless I get to play with SAV I would not know much about the interaction and looks like SAV is paid software unfortunately. 

But my guess is there is some security policy in SAV to disallow moving any interface to different namespace because then that interface becomes invisible to SAV and it cannot effectively monitor traffic in that interface. If so, you folks need to find a way to customize the SAV config to ignore docker created veth interfaces.

**Comment by typekpb** [comment]
Any idea how to disable SAV on RHEL (7.2)? thanks!

**Comment by thaJeztah** [comment]
@sw1sh is this still an issue or should this be closed?

**Comment by typekpb** [comment]
Well, I can confirm this is still an issue for IBM-ers (I'm running RHEL
7.2 with latest docker).

**Comment by thaJeztah** [comment]
@typekpb that's the SAV issue you're talking about? because that's not the original issue at the top here.

**Comment by typekpb** [comment]
Correct sav one.

**Comment by thaJeztah** [comment]
Alright, I'm going to close this issue because we need feedback fro @sw1sh if this is still an issue, and the SAV issue is unrelated to the original issue.

Given that the maintainers don't use SAV, perhaps someone inside IBM is able to look into that issue, and provide a PR? Note that apparently, the SAV issue is resolved for Ubuntu 15.04 (and newer?) https://github.com/docker/docker/issues/14904#issuecomment-127174269

**Comment by typekpb** [comment]
just documenting for my own reference, if you run on RHEL, disabling sav can be as easy as:
* updating relevant line in: `/etc/sysconfig/ibm-sav` and
* rebooting afterwards
